lib/Target/AMDGPU/SILowerControlFlow.cpp

```diff
@@ -230,8 +234,10 @@ void SILowerControlFlow::emitIf(MachineInstr &MI) {
 
   // Insert a pseudo terminator to help keep the verifier happy. This will also
   // be used later when inserting skips.
-  MachineInstr *NewBr = BuildMI(MBB, I, DL, TII->get(AMDGPU::SI_MASK_BRANCH))
-                            .add(MI.getOperand(2));
+  MachineInstr *NewBr = skipMaskBranch(MI, *MI.getParent());
```

Comment:
Try to insert the skip branch instruction here. If fails, insert the SI_MASK_BRANCH for usual check later during SIInsertSkips